api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -134,6 +134,21 @@ message RateLimit {
   // Optional additional prefix to use when emitting statistics. This allows to distinguish
   // emitted statistics between configured ``ratelimit`` filters in an HTTP filter chain.
   string stat_prefix = 13;
+
+  // If true, rate limit requests will also be sent to the rate limit service when the stream completes.
+  // This is useful when the rate limit budget needs to reflect the response context that is not available
+  // on the request path.
+  //
+  // For example, let's say the upstream service calculates the usage statistics, returns them in the response body
+  // and we want to utilize these numbers to apply the rate limit action for the subsequent requests.
+  // Combined with another filter that can set the desired addend based on the response (e.g. Lua filter),
+  // this can be used to subtract the usage statistics from the rate limit budget.
+  //
```

Comment:
on the response path, will the same `route.rate_limis_action` be matched again so the same descriptors will be sent in the RLS request ?

Reply:
in the current (poc) impl, yes.  hopefully  we could do better as in https://github.com/envoyproxy/envoy/pull/37567 by @wbpcode

Reply:
just to clarify though: `route.rate_limits.action` is not used for "matching" - it's just a meta data where the filter gets the addend from (defaults to 1)

Reply:
cool, this is more in line with the expectation I have as a user, would be great if it can be made more explicit in the comments

Reply:
referring to `expect_match` within it https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit-action-headervaluematch